engine/core/src/main/java/org/datacleaner/job/builder/TransformerComponentBuilder.java

```diff
@@ -334,7 +334,9 @@ protected void onRemovedInternal() {
      *            a new change listener
      */
     public void addChangeListener(TransformerChangeListener listener) {
-        _localChangeListeners.add(listener);
+        if (!_localChangeListeners.contains(listener)) {
```

Comment:
I checked and I wasn't ping-ponged here. It is logical that it's scary, because it's a poor man's solution to the problem that the addChangeListener is in fact invoked way too often (as a result of this pull request) and this is the only way too ensure the number of listeners doesn't completely explode. With the current setup I could figure out no way to neatly remove listeners when they're no longer needed without going through a number of extremely complex setup hoops.